### PR TITLE
Detector: Deletion of nested mapping

### PR DESCRIPTION
**Pull request by TilakMaddy**

### CODE REVIEW

File: tests/contract-playground/src/DeletionNestedMappingStructureContract.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        uint256[] names;
+        mapping(address => uint256) age;
+    }
+
+    mapping(address => Person) private people;
+
+    function remove() internal{
+        // We are deleting from a mapping whose value is a struct which contains a member of type mapping.
+        // Therefore, capture it!
+        delete people[msg.sender];
```
Comment (TilakMaddy) [inline tests/contract-playground/src/DeletionNestedMappingStructureContract.sol:17]: @alexroan Is my understanding here correct? If so, this PR is ready to review!

File: aderyn_core/src/detect/high/deletion_nested_mapping.rs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    fn severity(&self) -> IssueSeverity {
+        IssueSeverity::High
+    }
+
+    fn title(&self) -> String {
+        String::from("Deletion from a nested mappping.")
+    }
+
+    fn description(&self) -> String {
+        String::from("A deletion in a structure containing a mapping will not delete the mapping. The remaining data may be used to compromise the contract.")
```
Comment (TilakMaddy) [inline aderyn_core/src/detect/high/deletion_nested_mapping.rs:89]: "as-is" from Slither for parity

Comment (TilakMaddy) [reply aderyn_core/src/detect/high/deletion_nested_mapping.rs:89]: Can you help me understand the recommendation? I didn't quite get it : (

Comment (alexroan) [reply aderyn_core/src/detect/high/deletion_nested_mapping.rs:89]: Good question @TilakMaddy , it's not clear. My first thought was to just add a boolean `locked` field in the struct that you can check when retrieving the balances, but that would not prevent direct mapping references.

I'm not actually dure

**Review by alexroan — DISMISSED** [review]

**Review by alexroan — APPROVED** [review]

**Pull request merged**